### ISSUE TITLE
Pagination: hasPreviousPage unexpectedly returns true

### ISSUE THREAD

**Issue by jacobarriola** [wp-graphql/wp-graphql#1440]
**Repro steps:**

1. Run a search query to get first two results

2. Run a search query to get next two results (ie page 2)

3. Run a search query to get previous two results (ie back to initial results)

**Expected results**
* on query 3, `hasPreviousPage` should return `false`

**Actual results**
* on query 3, `hasPreviousPage` returns `true`

**Notes**
* everything works as expected when the `search` parameters are not included in the query

**Versions:**
WP-GraphQL: 0.10.3
WordPress: 5.4.2

#### Query 1 (initial search)
```graphql
query FirstTwoPosts {
  posts(first: 2, where: {search: "security"}) {
    pageInfo {
      endCursor
      startCursor
      hasPreviousPage
      hasNextPage
    }
    edges {
      cursor
    }
  }
}
```

#### Query 1 result 🆗 
```json
{
  "data": {
    "posts": {
      "pageInfo": {
        "endCursor": "YXJyYXljb25uZWN0aW9uOjg5MTI=",
        "startCursor": "YXJyYXljb25uZWN0aW9uOjYxMjIz",
        "hasPreviousPage": false,
        "hasNextPage": true
      },
      "edges": [
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjYxMjIz"
        },
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjg5MTI="
        }
      ]
    }
  }
}
```

#### Query 2 (page 2)
```graphql
query SecondTwoPosts {
  posts(first: 2, after: "YXJyYXljb25uZWN0aW9uOjg5MTI=", where: {search: "security"}) {
    pageInfo {
      endCursor
      startCursor
      hasPreviousPage
      hasNextPage
    }
    edges {
      cursor
    }
  }
}
```

#### Query 2 results 🆗 
```json
{
  "data": {
    "posts": {
      "pageInfo": {
        "endCursor": "YXJyYXljb25uZWN0aW9uOjU0MTQz",
        "startCursor": "YXJyYXljb25uZWN0aW9uOjIyNzI=",
        "hasPreviousPage": true,
        "hasNextPage": true
      },
      "edges": [
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjIyNzI="
        },
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjU0MTQz"
        }
      ]
    }
  }
}
```

#### Query 3 (go back to initial results)
```graphql
query PreviousTwoFromSecondGroup {
  posts(last: 2, before: "YXJyYXljb25uZWN0aW9uOjIyNzI=", where: {search: "security"}) {
    pageInfo {
      endCursor
      startCursor
      hasPreviousPage
      hasNextPage
    }
    edges {
      cursor
    }
  }
}
```

#### Query 3 result 💩 (`hasPreviousPage` is set to `true`; it should be `false` since we're at the beginning)
```json
{
  "data": {
    "posts": {
      "pageInfo": {
        "endCursor": "YXJyYXljb25uZWN0aW9uOjg5MTI=",
        "startCursor": "YXJyYXljb25uZWN0aW9uOjYxMjIz",
        "hasPreviousPage": true,
        "hasNextPage": true
      },
      "edges": [
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjYxMjIz"
        },
        {
          "cursor": "YXJyYXljb25uZWN0aW9uOjg5MTI="
        }
      ]
    }
  }
}
```

**Comment by izzygld** [comment]
Getting similar issue with using the `where` argument:

Example:

`query GetCategoryPosts(
	$where: RootQueryToCategoryConnectionWhereArgs!
	$first: Int
	$last: Int
	$after: String
	$before: String
) {
	categories(
		first: $first
		last: $last
		after: $after
		before: $before
		where: $where
	) {
		pageInfo {
			hasNextPage
			hasPreviousPage
			startCursor
			endCursor
		}
		edges {
			cursor
			node {
				name
				count
				posts {
					edges {
						node {
							title
						}
					}
				}
			}
		}
	}
}
`

**Results** - `hasNextPage` always returns false.


Removing the `where` argument - correct results are returned

**Comment by ardiewen** [comment]
@jacobarriola Have you seen the issue I created in #1411 yet? There appears to be a bug with backwards pagination.

For ordered lists, you can flip the order from ASC to DESC (or vice versa) during backwards pagination to get the correct results, but for your use case, you'll likely have to order your search results (e.g. add orderby: { field: ..., order: ASC/DESC } to your `where` parameter) to get this to work.

**Comment by jacobarriola** [comment]
Thanks @ardiewen . I'll give it a try!

**Comment by jasonbahl** [comment]
@jacobarriola @izzygld @ardiewen this is fixed in #1773. Please check it out and confirm things are working for you 🙏 

I'm going to add some tests to that PR specifically for this scenario to avoid regressions.

**Comment by jasonbahl** [comment]
@jacobarriola @izzygld @ardiewen I've included tests for this case in #1773. 

I'm going to close this as I believe it's resolved. If you feel like it's not fully resolved by #1773 or find other cases beyond what's tested in there, open another issue and reference this one. 

Thanks ya'll!
